### PR TITLE
Adds an achievement for pushing a boulder up a hill

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Credit to Melbert for this idea.
Adds a new achievement awarded for carrying a boulder from Lavaland to the Centcom Rescue Wing (via the Escape Shuttle).
If the boulder ever leaves your hands you will not gain the achievement.
If the boulder is not outside on Lavaland when you pick it up you will not gain the achievement.
If you arrive at the Rescue Ship (via pod) you will not gain the achievement.
If you drop the boulder 
If the shuttle is hijacked and you do not arrive at Centcom you will not gain the achievement (perhaps try hitting the hijacker with the boulder you are not allowed to drop?).
If you arrive at Centcom via the Escape Shuttle but fail to exit the shuttle holding the boulder (for instance, because a max cap bomb is detonated next to you one second after the shuttle arrives) you will not gain the achievement.

As the achievement is rewarded after the round-end report is generated, you also will not be represented on the round-end report. This is a personal struggle. 

Also carrying a boulder slows you down. Dragging it already did, so I'm not sure why carrying one did not. It feels like an oversight, and just makes sense.
People rarely have any particular need to manually carry them anywhere so this should be fine, right? 🙂 

## Why It's Good For The Game

Making players do pointless and weird bullshit is the reason to have an achievement system.
It's a social experiment.
I think that it is funny.

## Changelog

:cl: Jacquerel and Melbert
add: A new achievement for moving a boulder from one place to another, at great effort.
balance: Hauling a boulder around makes you move slower, as it does when dragging it.
/:cl:

### CODE REVIEW

**Review by MrMelbert — COMMENTED** [review]
Boulders have `drag_slowdown = 1.5`, it should also have a `slowdown = 1.5` (and `SLOWS_WHILE_IN_HAND`)

(I could do it in a separate pr myself if you wanna be atomic)

File: code/datums/components/sisyphus_awarder.dm [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/// Called when we're picked up, check if we're in the right place to start our epic journey
+/datum/component/sisyphus_awarder/proc/on_picked_up(atom/source, mob/living/the_taker)
+	SIGNAL_HANDLER
+	if (!istype(get_area(the_taker), /area/lavaland))
+		qdel(src)
+		return
+	UnregisterSignal(parent, COMSIG_ITEM_PICKUP)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, PROC_REF(on_dropped))
+	RegisterSignal(the_taker, COMSIG_ENTER_AREA, PROC_REF(on_bearer_changed_area))
+	sisyphus = WEAKREF(the_taker)
```
Comment (LemonInTheDark) [inline code/datums/components/sisyphus_awarder.dm:32]: why is this a weakref? we delete if the item is ever dropped. could do it on moved IG? would be more safe

**Comment by breadcrombs** [comment]
sisyphus in ss13 confirmed

File: code/datums/components/sisyphus_awarder.dm (hunk 1 above)
Comment (Jacquerel) [reply code/datums/components/sisyphus_awarder.dm:32]: yeah I guess if it's always held it's probably reasonably safe to assume that we'll delete alongside the person holding us
I'll change it to moved and use a regular reference

File: code/datums/components/sisyphus_awarder.dm [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/**
+ * This component awards the sisyphus achievement if you cart a boulder from lavaland to centcom
+ * It's not really reusable but its a component just to encapsulate and destroy the behaviour neatly
+ */
+/datum/component/sisyphus_awarder
+	/// What poor sap is hauling this rock?
+	var/datum/weakref/sisyphus
+
+/datum/component/sisyphus_awarder/Initialize()
+	. = ..()
```
Comment (MrMelbert) [inline code/datums/components/sisyphus_awarder.dm:10]:
```suggestion
```

**Comment by SomeoneYouProbablyKnow** [comment]
Can you add the audio sting for when you carry out the boulder out of the shuttle? Or is that too "meme of the week"

**Comment by Jacquerel** [comment]
> Can you add the audio sting for when you carry out the boulder out of the shuttle? Or is that too "meme of the week"

i dont know what the audio sting is but sisyphus is from ancient greece

**Comment by SomeoneYouProbablyKnow** [comment]
I kinda thought this was in reference to the sisyphus meme (which really is just the actual ancient greek myth returning to the zeitgeist). And the meme has its own i guess associated sound? song? Leitmotif? This thing
https://youtu.be/jcYvDZKLQJ0?si=BX86MbgSLxql_pFo
But as i said it might be not be great idea to add it to the actual game

**Comment by Jacquerel** [comment]
im probably not going to do that yeah

**Comment by SomeoneYouProbablyKnow** [comment]
Fair nuff

**Comment by DaCoolBoss** [comment]
We need more stuff like this. Very nice.

**Review by MrMelbert — APPROVED** [review]

**Pull request merged**